allauth/socialaccount/tests.py

```diff
@@ -229,13 +305,26 @@ def login(self, resp_mock, process="login", with_refresh_token=True):
             + "?"
             + urlencode(dict(process=process))
         )
+
         p = urlparse(resp["location"])
         q = parse_qs(p.query)
+
+        pkce_enabled = app_settings.PROVIDERS.get(self.provider_id, {}).get(
+            "OAUTH_PKCE_ENABLED", False
+        )
+
+        self.assertEqual("code_challenge" in q, pkce_enabled)
+        self.assertEqual("code_challenge_method" in q, pkce_enabled)
+        if pkce_enabled:
+            code_challenge = q["code_challenge"][0]
```

Comment:
Should there be an assertion after this?

Reply:
What assertion are you thinking of?

Reply:
No, the assertion comes down below! The assertion is that the `code_challenge` that's provided here matches the hashed `code_verifier` that's provided below. So this is just assigning the code challenge value to a variable.

Reply:
No, the assertion comes down below! The assertion is that the `code_challenge` that's provided here matches the hashed `code_verifier` that's provided below. So this is just assigning the code challenge value to a variable.